### PR TITLE
Ray Launcher Plugin support for Python 3.11

### PR DESCRIPTION
**Pull request by Pedrexus**
## Motivation

Ray Launcher Plugin is now being supported in python 3.11 and Ray 2.7

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebookresearch/hydra/blob/main/CONTRIBUTING.md)?

Yes

## Test Notes

The tests are now passing. Updating the libraries to the newer version solved most issues.

AWS Ray Launcher is being deprecated as we could not execute tests within AWS infrastructure.

## Related Issues and PRs

#2489

### CODE REVIEW

**Comment by Jasha10** [comment]
Looks some of the test failures are due to a deprecation warning.
For example, on the `test_linux-3.10` ci job:
```
FAILED tests/test_submitit_launcher.py::TestSubmititLauncher::test_sweep_1_job_strict[submitit_local-overrides0] - DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
```

We are already ignoring this deprecation warning for the core tests; see [line 511 of the noxfile](https://github.com/facebookresearch/hydra/blob/809718cdcd64f9cd930d26dea69f2660a6ffa833/noxfile.py#L511).
I think it should be possible to do the same thing for plugin tests by modifying line 581 of the noxfile:
```diff
diff --git a/noxfile.py b/noxfile.py
index 693717dfe5..7a62ed1026 100644
--- a/noxfile.py
+++ b/noxfile.py
@@ -578,7 +578,11 @@ def test_selected_plugins(session: Session, selected_plugins: List[Plugin]) -> N
     # Run tests for all installed plugins
     for plugin in selected_plugins:
         session.chdir(plugin.abspath)
-        run_pytest(session)
+        run_pytest(
+            session,
+            ".",
+            "-W ignore:pkg_resources is deprecated as an API:DeprecationWarning",
+        )
 
 
 @nox.session(python="3.8")  # type: ignore
```

**Comment by Pedrexus** [comment]
it seems that adding "-W ignore:pkg_resources is deprecated as an API:DeprecationWarning" did not work. Other libraries do not have their tests passing.

**Comment by Jasha10** [comment]
Still a few more linter issues:
```
nox > [2023-12-25 05:04:39,096] Command isort . --check --diff failed with exit code 1:
ERROR: /home/circleci/project/plugins/hydra_ray_launcher/tests/test_ray_aws_launcher.py Imports are incorrectly sorted and/or formatted.
--- /home/circleci/project/plugins/hydra_ray_launcher/tests/test_ray_aws_launcher.py:before     2023-12-25 04:52:13.966308
+++ /home/circleci/project/plugins/hydra_ray_launcher/tests/test_ray_aws_launcher.py:after      2023-12-25 05:04:39.064854
@@ -6,11 +6,11 @@
 import subprocess
 import sys
 import tempfile
+from importlib.metadata import version
 from pathlib import Path
 from typing import Any, Dict, Generator, Optional
 
 import boto3  # type: ignore
-from importlib.metadata import version
 from botocore.exceptions import NoCredentialsError, NoRegionError  # type: ignore
 from hydra.core.plugins import Plugins
 from hydra.plugins.launcher import Launcher
Skipped 1 files
nox > [2023-12-25 05:04:39,096] Session lint_plugins-3.11(hydra-ray-launcher) failed.
```

**Comment by Jasha10** [comment]
> it seems that adding "-W ignore:pkg_resources is deprecated as an API:DeprecationWarning" did not work. Other libraries do not have their tests passing.

I'll submit a PR to address this shortly.

**Comment by Pedrexus** [comment]
I was able to fix the lint issue via `pre-commit run --files ./plugins/**/*`

**Comment by bryant1410** [comment]
Hey, I just wanted to share that some of these changes overlap with those in #2713, which I believe introduces a simpler change that can be merged already (I've been using that branch fine for a while), and then move to those in this PR. What do you think?

File: plugins/hydra_ray_launcher/setup.py [hunk 1]
```diff
@@ -20,16 +20,18 @@
...
         "Operating System :: POSIX :: Linux",
     ],
     install_requires=[
-        "boto3==1.22.6",
-        "hydra-core>=1.1.2",
-        "ray[default]~=1.12",
-        "aiohttp==3.8.1",
-        "cloudpickle==2.0.0",
-        "pickle5==0.0.11; python_version < '3.8'",
+        "boto3>=1.28,<1.29",
```
Comment (bryant1410) [inline plugins/hydra_ray_launcher/setup.py:29]: `boto3` is constantly updating minor versions, and this constraint may be too harsh. What about the following?

```suggestion
        "boto3>=1.28,<2",
```

**Comment by Pedrexus** [comment]
@Jasha10 should we close this issue given #2713  has been merged?

**Comment by Jasha10** [comment]
Yes. Thank you @Pedrexus.

**Pull request closed without merging**